packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/HeatmapController.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.googlemaps;
+
+import androidx.annotation.NonNull;
+import com.google.android.gms.maps.model.TileOverlay;
+import com.google.maps.android.heatmaps.Gradient;
+import com.google.maps.android.heatmaps.HeatmapTileProvider;
+import com.google.maps.android.heatmaps.WeightedLatLng;
+import java.util.List;
+
+/** Controller of a single Heatmap on the map. */
+public class HeatmapController implements HeatmapOptionsSink {
+  private final @NonNull HeatmapTileProvider heatmap;
+  private final @NonNull TileOverlay heatmapTileOverlay;
+
+  /** Construct a HeatmapController with the given heatmap and heatmapTileOverlay. */
+  HeatmapController(@NonNull HeatmapTileProvider heatmap, @NonNull TileOverlay heatmapTileOverlay) {
+    this.heatmap = heatmap;
+    this.heatmapTileOverlay = heatmapTileOverlay;
+  }
+
+  /** Remove the heatmap from the map. */
```

Comment:
Is this comment correct? The method is on HeatMapController, it looks like the the heatmap is a provider but what is removed is the tileOverlay. I am not sure this dart doc makes sense to someone coming into the this code.

Reply:
If this comment is incorrect then the code is not working. The Google Maps tile overlay stuff is confusing to look at. The `HeatmapTileProvider` is what provides the data that `heatmapTileOverlay` shows on the map. So removing `heatmapTileOverlay` removes that data from the map display.